components/calendar/src/provider.rs

```diff
@@ -151,3 +151,44 @@ pub struct WeekDataV1 {
     /// For a given week, the minimum number of that week's days present in a given month or year for the week to be considered part of that month or year.
     pub min_week_days: u8,
 }
+
+/// Bitset representing weekdays that are part of the 'weekend'.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_calendar::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+pub struct WeekendSet(pub u8);
```

Comment:
`nit`: please, don't make `u8` public. It's a new [type idiom](https://doc.rust-lang.org/rust-by-example/generics/new_types.html) and we need to validate it. You have one bit free, and we want to keep it this way (it may come useful), you need to control endianness, etc.

I suggest to set quite clear API for this with a constructor, and getter for now. And I suggest the getter to be not returning a bitmap, but rather something like `Box<u8[]>` or an iterator (to avoid preallocating by returning `Vec<u8>`).

Reply:
Good one, thanks!
Addressed in https://github.com/unicode-org/icu4x/pull/4364/commits/db654ead02b555c34aba8927df4956c2fd7d0c9e.